lib/qa/authorities/assign_fast/generic_authority.rb

```diff
@@ -0,0 +1,58 @@
+module Qa::Authorities
+  # A wrapper around the FAST api for use with questioning_authority
+  # API documentation:
+  # http://www.oclc.org/developer/develop/web-services/fast-api/assign-fast.en.html
+  class AssignFast::GenericAuthority < Base
+    attr_reader :subauthority
+    def initialize(subauthority)
+      @subauthority = subauthority
+    end
+
+    include WebServiceBase
+
+    # Search the FAST api
+    #
+    # @param [String] the query
+    # @return json results
+    def search q
+      begin
+        @raw_response = get_json(build_query_url(q))
+      rescue JSON::ParserError => error
+        # TODO: log the url and the error
+        return []
+      end
+      parse_authority_response
+    end
+
+    # Build a FAST API url
+    #
+    # @param [String] the query
+    # @return [String] the url
+    def build_query_url q
+      # certain chars not tolerated by the API
+      q.gsub! /-|\(|\)|:/, ""
+      escaped_query = URI.escape(q)
+      index = AssignFast.get_index_for_authority(subauthority)
+      return_data = "#{index}%2Cidroot%2Cauth%2Ctype"
+      num_rows = 20 # max allowed by the API
+      url = "http://fast.oclc.org/searchfast/fastsuggest?&query=#{escaped_query}&queryIndex=#{index}&queryReturn=#{return_data}&suggest=autoSubject&rows=#{num_rows}"
+    end
+
+    private
+
+    def parse_authority_response
+      r = @raw_response
```

Comment:
In this case, I'd suggest passing the `@raw_response` as a parameter to the method. That would make this method more testable.